frontend/simple/views/components/Chatroom/SendArea.vue

```diff
@@ -0,0 +1,177 @@
+      pointer-events: initial;
+    }
+  }
+}
+</style>
+<script>
+export default {
+  name: 'Chatroom',
+  components: {},
+  props: {
+    title: String,
+    searchPlaceholder: String
+  },
+  data () {
+    return {
+      config: {
+        sendPlaceholder: [this.L('Be nice to'), this.L('Be cool to'), this.L('Have fun with')]
+      },
+      ephemeral: {
+        actionsWidth: '',
+        maskHeight: '',
+        textWithLines: ''
+      }
+    }
+  },
+  mounted () {
+    // Get actionsWidth to add a dynamic padding to textarea,
+    // so those actions don't be above the textarea's value
+    this.ephemeral.actionsWidth = this.$refs.actions.offsetWidth
+    this.updateTextArea()
+  },
+  computed: {
+    textareaStyles () {
+      return {
+        paddingRight: this.ephemeral.actionsWidth + 'px',
+        height: this.ephemeral.maskHeight + 'px'
+      }
+    },
+    maskStyles () {
+      return {
+        paddingRight: this.ephemeral.actionsWidth + 'px'
+      }
+    },
+    isVisible () {
+      // REVIEW - should it always be visible or only when it has text?
+      return true // !!this.ephemeral.textWithLines
+    },
+    customSendPlaceholder () {
+      return `${this.config.sendPlaceholder[Math.floor(Math.random() * this.config.sendPlaceholder.length)]} ${this.title}`
+    }
+  },
+  methods: {
+    handleKeydown (e) {
+      const enterKey = e.keyCode === 13
+
+      if ((e.shiftKey || e.altKey || e.ctrlKey) && enterKey) {
+        return this.createNewLine()
+      } else if (enterKey) {
+        return this.sendMessage()
+      } else {
+        this.updateTextArea()
+      }
+    },
+    updateTextWithLines () {
+      this.ephemeral.textWithLines = this.$refs.textarea.value.replace(/\n/g, '<br>')
+    },
+    updateTextArea () {
+      this.updateTextWithLines()
+
+      const length = this.ephemeral.textWithLines.length
+      const isLastLineEmpty = this.ephemeral.textWithLines.substring(length - 4, length) === '<br>'
+
+      // TRICK: Use an invisible element (.mask) as placeholder to know the
+      // amount of space the user message takes... (taking in account new lines)
+      this.$refs.mask.innerHTML = this.ephemeral.textWithLines + (isLastLineEmpty ? '.' : '')
+
+      // ...and apply the maks's height to the textarea so it dynamically grows as the user types
+      this.ephemeral.maskHeight = this.$refs.mask.offsetHeight
+
+      // ... finaly inform the parent about the new height to adjust the layout
+      this.$emit('heightUpdate', this.ephemeral.maskHeight + 'px')
+    },
+    createNewLine () {
+      this.$refs.textarea.value += '\n'
+      this.updateTextArea()
+    },
+    sendMessage () {
+      if (!this.ephemeral.textWithLines) {
+        return false
+      }
+
+      this.updateTextWithLines()
+      this.$emit('send', this.ephemeral.textWithLines) // TODO remove first / last lines when empty
+      this.$refs.textarea.value = ''
+      this.updateTextArea()
+    }
+  }
+}
+</script>
```

Comment:
Hmm, I noticed that when pressing shift+enter in safari to expand vertically the size of the text area, that I could press delete to delete paragraphs and it wouldn't shrink immediately down to size, but it would take some time before that happened. Weird.

Reply:
I'll take a look at that and let you know when I have news about it

Reply:
The same happened in the chrome too, fixed.

The problem was the height calculation was happening on keydown for both shift+enter and new characters (ex: delete). But the height calculation for new characters should happen on keyup, otherwise the new value would exclude the last character inserted. That's why the height calculation is incorrect if it's one character away of changing heights.